### ISSUE TITLE
lint: ptrToRefParam probably should not check underlying types

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#262]
Reproducer:
```
gocritic check-package -enable=ptrToRefParam go/types
```
Output:
```
src/go/types/typexpr.go:486:36: ptrToRefParam: consider `oset' to be of non-pointer type
```
Code:
```go
func (check *Checker) declareInSet(oset *objset, pos token.Pos, obj Object) bool {
```
And here is `objset` definition:
```go
type objset map[string]Object
```
To reduce the amount of false positives, ptrToRefParam should probably ignore underlying types.

Negative tests suite for this checker should be also expanded.
